std/encoding.d

```diff
@@ -378,65 +378,65 @@ template EncoderFunctions()
     template ReadFromString()
     {
         @property bool canRead() { return s.length != 0; }
-        E peek() { return s[0]; }
-        E read() { E t = s[0]; s = s[1..$]; return t; }
+        E peek()() { return s[0]; }
+        E read()() { E t = s[0]; s = s[1..$]; return t; }
```

Comment:
Hmm, good point. SInce these functions are inside a template, shouldn't the compiler infer attributes? Is there an issue filed for this?